### DISCUSSION TITLE
Podman Quadlet Keycloak - Unable to start container due to parameter in entrypoint

### DISCUSSION THREAD

**Discussion by aitzubi** [containers/podman#21240] [Q&A] [answered]
Hi,

I am trying to use quadlets to run a Keycloak container. In order to start the keycloak container, you need to override the entrypoint parameter with something similar to this:

`--entrypoint /opt/keycloak/bin/kc.sh start`

The issue is that I am unable to translate this to systemd. I`ve tried using the Entrypoint option in the Container section in systemd and the PodmanArgs option but no luck. It seems that systemd is unable to recognize the aditional parameter (start).

Some examples:

```
PodmanArgs=--entrypoint /opt/keycloak/bin/kc.sh start
PodmanArgs=--entrypoint \"/opt/keycloak/bin/kc.sh start\"

EntryPoint=/opt/keycloak/bin/kc.sh start
EntryPoint=\"/opt/keycloak/bin/kc.sh start\"
```

If I remove the start parameter, the container runs without any issue but the keycloak app does not run.

Can anybody help me with this?

Kind regards

**Comment by aitzubi** [comment]
Forgot to say that if I run (podman run) the container with that entrypoint (--entrypoint /opt/keycloak/bin/kc.sh start) manually, it works as expected

**Comment by ygalblum** [comment]
Thanks for bringing this up.
Can you please provide the complete `.container` file that fails and the manual command the works?
I've tried looking for an example, but the docs I found (for example https://www.keycloak.org/server/containers) do not require overriding the entrypoint.
In addition, have you tried setting `Exec` instead of `Entrypoint`?

**Comment by jorti** [comment, marked as answer]
You need to define both a entrypoint with `PodmanArgs` and use `Exec` for the arguments:

```
PodmanArgs=--entrypoint=/opt/keycloak/bin/kc.sh
Exec=start
```

**Comment by aitzubi** [comment]
It works! Thank you guys!
